editorextensions/codebeamer-custom-ui/modal/src/api/lucidGateway.ts

```diff
@@ -5,11 +5,77 @@ import { store } from '../store/store';
 import TrackerDetails from '../models/trackerDetails.if';
 import { CardData } from '../models/lucidCardData';
 
+/**
+ * Class for handling message events and callbacks.
+ */
+export class MessageHandler {
+	private callbacks: ((data: any) => void)[] = [];
+
+	/**
+	 * Creates an instance of MessageHandler and sets up a message event listener.
+	 */
+	constructor() {
+		window.addEventListener('message', (e) => {
+			const data = JSON.parse(e.data);
+			this.notifyCallbacks(data);
+		});
+	}
```

Comment:
Very nice 👍🏽 
I think we should use an additional pattern here, namely the [Singleton](https://refactoring.guru/design-patterns/singleton) - we only really want one instance of the `MessageHandler` when / if in future we subscribe to messages from other sources.

Reply:
did not think of this, very cool